src/main.rs

```diff
@@ -15,17 +15,49 @@
 use jujutsu::cli_util::{create_ui, handle_command_result, parse_args, CommandError};
 use jujutsu::commands::{default_app, run_command};
 use jujutsu::ui::Ui;
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::reload::Handle;
+use tracing_subscriber::EnvFilter;
 
-fn run(ui: &mut Ui) -> Result<(), CommandError> {
+fn run(
+    ui: &mut Ui,
+    reload_log_filter: Handle<EnvFilter, impl tracing::Subscriber>,
+) -> Result<(), CommandError> {
     let app = default_app();
     let (command_helper, matches) = parse_args(ui, app, std::env::args_os())?;
+    if command_helper.global_args().verbose {
+        reload_log_filter
+            .modify(|filter| {
+                *filter = EnvFilter::builder()
+                    .with_default_directive(LevelFilter::DEBUG.into())
+                    .from_env_lossy()
+            })
+            .map_err(|err| {
+                CommandError::InternalError(format!("failed to enable verbose logging: {:?}", err))
+            })?;
+        tracing::debug!("verbose logging enabled");
+    }
     run_command(ui, &command_helper, &matches)
 }
 
 fn main() {
+    // TODO(@rslabbert): restructure logging filter setup to better handle
+    // having verbose logging set up as early as possible, and to support
+    // custom commands. See discussion on:
+    // https://github.com/martinvonz/jj/pull/771
+    let filter = EnvFilter::builder()
+        .with_default_directive(LevelFilter::INFO.into())
+        .from_env_lossy();
+    let (filter, reload_log_filter) = tracing_subscriber::reload::Layer::new(filter);
+    tracing_subscriber::registry()
+        .with(filter)
+        .with(tracing_subscriber::fmt::Layer::default())
```

Comment:
Maybe better to use stderr instead?
https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_writer

Reply:
Good shout, I've updated it to use stderr, which should make it more reliable in piping and such workflows.

Reply:
thanks!